### PR TITLE
make metadata processing more robust

### PR DESCRIPTION
**Pull request by yymao**
@evevkovacs This is a PR to your PR https://github.com/LSSTDESC/gcr-catalogs/pull/284; I am attempting to make the metadata processing (`box_size` and `redshift`) more robust, particularly for the case when the metadata are missing in the files.

### CODE REVIEW

**Review by evevkovacs — APPROVED** [review]
I didn't test it but I read through everything and it looks ok. One question: in the process_metadata function, the code only looks for either 'sky_area' or ('redshift', box_size'). Is there somewhere else that reads in all the metadata available in the catalog and saves it into the catalog instance?  Or is the philosophy that the catalog provider should decide which metadata should be made available to the user via the process_metadata function?

**Comment by yymao** [comment]
We can indeed import more metadata (and that's indeed part of the reason why I change to this more generic construct). I actually don't know what metadata are available in the catalog. 

But if we do want to import more metadata, we also need to provide default values. 

Once you are happy with this PR, can you merge it so that it'll reflected in your original PR? Thanks!

**Pull request merged**
